tests/mapreduce/test.persisted.js

```diff
@@ -243,6 +243,21 @@ function tests(suiteName, dbName, dbType) {
       }).should.be.rejected;
     });
 
+    it("Query non-string view returns error", function () {
+      var db = new PouchDB(dbName);
+      var doc = {
+        _id: '_design/barbar',
+        views: {
+          scores: {
+            map: 1
+          }
+        }
+      };
+      return db.post(doc).then(function () {
+        return db.query('barbar/scores', {key: 'bar'});
+      }).should.be.rejected;
```

Comment:
This looks like it would incorrectly pass if `db.post()` failed (e.g. if we add validation to reject invalid design docs similar to CouchDB). Should the test be more explicit about which promise is rejected?